rosbag2_cpp/include/rosbag2_cpp/writer_interfaces/base_writer_interface.hpp

```diff
@@ -45,6 +45,8 @@ class ROSBAG2_CPP_PUBLIC BaseWriterInterface
   virtual void remove_topic(const rosbag2_storage::TopicMetadata & topic_with_type) = 0;
 
   virtual void write(std::shared_ptr<rosbag2_storage::SerializedBagMessage> message) = 0;
+
+  virtual void take_snapshot() = 0;
```

Comment:
This needs a docstring please.

Maybe it should return a `bool`? If a Writer is not configured to do snapshot-writing, then it would make sense to provide some indication that "no i won't do this" or "nothing happens"